hack/kind-install-for-capd.sh

```diff
@@ -32,22 +32,29 @@ fi
 
 KIND_CLUSTER_NAME=${CAPI_KIND_CLUSTER_NAME:-"capi-test"}
 
+
+# 1. If kind cluster already exists exit.
 if [[ "$(kind get clusters)" =~ .*"${KIND_CLUSTER_NAME}".* ]]; then
   echo "kind cluster already exists, moving on"
   exit 0
 fi
 
-# create registry container unless it already exists
+# 2. Create registry container unless it already exists
 reg_name='kind-registry'
 reg_port='5000'
-running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
-if [ "${running}" != 'true' ]; then
+if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
   docker run \
     -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
     registry:2
 fi
 
-# create a cluster with the local registry enabled in containerd
+# 3. Create kind cluster with containerd registry config dir enabled.
+# TODO(killianmuldoon): kind will eventually enable this by default and this patch will be unnecessary.
```

Comment:
Should we open an issue for the TODO as well? Issues are harder to miss then TODO's somewhere in our code (when the stale bot updates them)

Reply:
Will do

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/8736